### PR TITLE
action: add cargo deny support

### PR DESCRIPTION
**Pull request by bergwolf**
Use it to scan code license and security issues.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -32,16 +32,20 @@ serde_with = { version = "1.6.0", features = ["macros"] }
...
 vm-memory = { version = "0.7.0", features = ["backend-mmap"], optional = true }
 chrono = "0.4.19"
-openssl = { version = "0.10.35", features = ["vendored"] }
+openssl = { version = "0.10.38", features = ["vendored"] }
 hyperlocal = "0.8.0"
-tokio = { version = "1.9.0", features = ["macros"] }
+tokio = { version = ">=1.13.1", features = ["macros"] }
 hyper = "0.14.11"
+# pin openssl-src to bring in fix for RUSTSEC-2021-0098
+openssl-src = ">=111.16.0"
```
Comment (changweige) [inline Cargo.toml:46]: Well, those dependencies updates were suggested by Cargo Deny?

Comment (bergwolf) [reply Cargo.toml:46]: yup

File: rafs/Cargo.toml [hunk 2]
```diff
@@ -3,10 +3,12 @@ name = "rafs"
 version = "0.1.0"
 authors = ["The Nydus Developers"]
 edition = "2018"
+license = "Apache-2.0 OR BSD-3-Clause"
 
 [dependencies]
 anyhow = "1.0.35"
-arc-swap = "0.4.6"
+# pin arc-swap because 1.x version of ArcSwapAny does not implement Clone
+arc-swap = "=0.4"
```
Comment (changweige) [inline rafs/Cargo.toml:11]: But this patch still implements `Clone` itself?

File: storage/src/device.rs [hunk 3]
```diff
@@ -799,12 +799,19 @@ pub trait BlobObject: AsRawFd {
 ///
 /// All blob Io requests are actually served by the underlying [BlobCache] object. A new method
 /// [update()]() is added to switch the storage backend on demand.
-#[derive(Clone)]
 pub struct BlobDevice {
     //meta: ArcSwap<Arc<dyn BlobCache>>,
     blobs: ArcSwap<Vec<Arc<dyn BlobCache>>>,
     blob_count: usize,
 }
+impl Clone for BlobDevice {
```
Comment (changweige) [inline storage/src/device.rs:807]: I am curious why the `arc-library` eliminated the Clone implementation but we can safely implement by ourselves?

File: rafs/Cargo.toml (hunk 2 above)
Comment (bergwolf) [reply rafs/Cargo.toml:11]: Ah, my bad. Should have dropped the Clone implementation. Fixed.

**Review by changweige — APPROVED** [review]
LGTM!

File: Cargo.toml [hunk 4]
```diff
@@ -3,6 +3,7 @@ name = "nydus-rs"
 version = "2.0.0-rc.0"
 authors = ["The Nydus Developers"]
 edition = "2018"
+license = "Apache-2.0 OR BSD-3-Clause"
```
Comment (jiangliu) [inline Cargo.toml:6]: The BSD-3-Clause need special treatment, and seems we have only Apache 2.0 license file attached.

File: LICENSE-BSD-3-Clause [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//    * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//    * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//    * Neither the name of Google Inc. nor the names of its
```
Comment (hsiangkao) [inline LICENSE-BSD-3-Clause:13]: Google Inc.?

https://opensource.org/licenses/BSD-3-Clause

Comment (bergwolf) [reply LICENSE-BSD-3-Clause]: Good catch, thanks! That one was copied from rust-vmm :(

Comment (jiangliu) [reply LICENSE-BSD-3-Clause]: That's where we need to tweak the BSD-3-Clause license:(

**Review by jiangliu — APPROVED** [review]

**Pull request merged**